devdoc/worker_thread_requirements.md

```diff
@@ -125,21 +119,20 @@ MOCKABLE_FUNCTION(, WORKER_THREAD_SCHEDULE_PROCESS_RESULT, worker_thread_schedul
 
 **SRS_WORKER_THREAD_01_042: [** If `sm_exec_begin` does not grant the execution, `worker_thread_schedule_process` shall fail and return `WORKER_THREAD_SCHEDULE_PROCESS_INVALID_STATE`. **]**
 
-**SRS_WORKER_THREAD_01_017: [** `worker_thread_schedule_process` shall set the process event created in `worker_thread_create`. **]**
+**SRS_WORKER_THREAD_01_017: [** `worker_thread_schedule_process` shall set the state to `WORKER_THREAD_STATE_PROCESS_ITEM`. **]**
 
 **SRS_WORKER_THREAD_01_043: [** `worker_thread_schedule_process` shall call `sm_exec_end`. **]**
 
 **SRS_WORKER_THREAD_01_015: [** On success `worker_thread_schedule_process` shall return `WORKER_THREAD_SCHEDULE_PROCESS_OK`. **]**
 
-**SRS_WORKER_THREAD_01_018: [** If any other error occurs, `worker_thread_schedule_process` shall fail and return `WORKER_THREAD_SCHEDULE_PROCESS_ERROR`. **]**
-
 ### worker_thread
 
-**SRS_WORKER_THREAD_01_019: [** The worker thread started by `worker_thread_create` shall wait for the 2 events: thread shutdown event and execute worker function event. **]**
+**SRS_WORKER_THREAD_01_019: [** The worker thread started by `worker_thread_create` shall get the module state. **]**
 
-**SRS_WORKER_THREAD_01_020: [** When the shutdown event is signaled, the worker thread shall exit. **]**
+**SRS_WORKER_THREAD_01_020: [** If the module state is `WORKER_THREAD_STATE_CLOSE`, the worker thread shall exit. **]**
```

Comment:
>WORKER_THREAD_STATE_CLOSE [](http://example.com/codeflow?start=56&length=25)

likely should be explained in some enum or something.

Reply:
I may know know what you're meaning here, but these are already enums.